src/Aspire.Dashboard/Components/Controls/Chart/MetricTable.razor.cs

```diff
@@ -81,6 +88,11 @@ protected override async Task OnChartUpdated(List<ChartTrace> traces, List<DateT
         for (var i = 0; i < xValues.Count; i++)
         {
             var xValue = xValues[i];
+            if (_metrics.TryGetValue(xValue, out var oldMetric))
+            {
+                newMetrics.Add(xValue, oldMetric);
+                continue;
+            }
```

Comment:
Currently, the behavior of `FluentDataGrid` is such that if you have focus on a row i (backed by some object items[i]) and then items[i] changes, focus is un-set. The current behavior of the metric table update function is to create new objects for each point on each update, which will result in focus being un-set within 200ms of focusing a row (since that's the time between instrument updates). 

If we preserve the old `MetricViewBase` object (since the data for the given timestamp would not have changed), focus will not be lost since the object will not have changed.

Reply:
And are dimensions a factor of metric? Can we have an update on multiple dimensions for the metrics at the same time? In that case we would want the focus to be respected - or am I overthinking this :)